apps/condo/domains/meter/schema/MeterReadingExportTask.test.js

```diff
@@ -0,0 +1,519 @@
+
+            test('cannot delete', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await MeterReadingExportTask.delete(anonymous, adminTask.id)
+                })
+            })
+
+            test('cannot soft-delete', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await MeterReadingExportTask.softDelete(anonymous, adminTask.id)
+                })
+            })
+        })
+    })
+
+    describe('Validation tests', () => {
+        test('should throw validation error if you trying to change status of already completed task', async () => {
+            const { organization, userClient } = await makeAdminClientWithRegisteredOrganizationWithRoleWithEmployee({
+                canReadMeters: true,
+                canManageMeters: true,
+                canManageMeterReadings: true,
+            })
+            const [task] = await createTestMeterReadingExportTask(userClient, userClient.user, {
+                where: {
+                    organization: {
+                        id: organization.id,
+                    },
+                },
+            })
+
+            await waitFor(async () => {
+                const updatedTask = await MeterReadingExportTask.getOne(userClient, {
+                    id: task.id,
+                })
+                expect(updatedTask.status).toEqual(COMPLETED)
+            })
+
+            await expectToThrowGQLError(async () => {
+                await updateTestMeterReadingExportTask(userClient, task.id, { status: CANCELLED })
+            }, ERRORS.STATUS_IS_ALREADY_COMPLETED)
+        })
+    })
+})
+
+describe('exportMeterReadings', () => {
+    it('should create `MeterReadingExportTask` and create xlsx file', async () => {
+        const { organization, userClient } = await makeAdminClientWithRegisteredOrganizationWithRoleWithEmployee({
+            canManageProperties: true,
+            canManageMeters: true,
+            canManageMeterReadings: true,
+            canReadMeters: true,
+        })
+        const [property] = await createTestProperty(userClient, organization)
+        
+        const meterReadings = []
+
+        const [resource] = await MeterResource.getAll(userClient, { id: COLD_WATER_METER_RESOURCE_ID })
+        const [source] = await MeterReadingSource.getAll(userClient, { id: CALL_METER_READING_SOURCE_ID })
+
+        const metersCount = 1
+        const meterReadingsOnEachMeter = 1
+        const meterReadingsCount = meterReadingsOnEachMeter * metersCount
+
+        for (let i = 0; i < metersCount; i++) {
+            const [meter] = await createTestMeter(userClient, organization, property, resource, {})
+
+            for (let j = 0; j < meterReadingsOnEachMeter; j++) {
+                const [meterReading] = await createTestMeterReading(userClient, meter, source)
+
+                meterReadings.push(meterReading)
+            }
+        }
```

Comment:
Maybe increase the amount of data a little?
A loop within a loop for the sake of one record looks cumbersome :)

Reply:
Oh, 1 been for debug purposes) Will change it